web/src/engine/osk/src/input/gestures/specsForLayout.ts

```diff
@@ -522,7 +522,7 @@ export function longpressContactModel(params: FullGestureParams, enabledFlicks:
          * The 'indexOf' allows 'n', 'nw', and 'ne' - approx 67.5 degrees on
          * each side of due N in total.
          */
-        if((enabledFlicks && spec.permitsFlick(stats.lastSample.item)) && (stats.cardinalDirection?.indexOf('n') != -1 ?? false)) {
+        if((enabledFlicks && spec.permitsFlick(stats.lastSample.item)) && ((stats.cardinalDirection ?? '').indexOf('n') != -1)) {
```

Comment:
I was getting a new warning here about the optional-chaining being invalid - that it would never actually trigger.  Technically true, as `undefined != -1`... so I went ahead and fixed it while in the same file.

Reply:
do we have a test for this change? seems like it was buggy before but I've got no idea from here (a) what would be needed to trigger the bug, and (b) what the outcome was.

Reply:
What would be needed to trigger the bug:  the user would have to return the touchpoint to the exact starting pixel of the gesture.  It'd certainly be hard to do consistently by hand.

If anything could be triggered by the bad logic check, it'd be for the longpress menu to display early when it shouldn't - that's what the main body of the enabled block is intended to verify and do.  I believe this was prevented by lines 528 and 529, though - the same condition checked by the highlighted line above would also cause `.angle == NaN`, leading to `NaN > spec.flickDistFinal` - and conditionals against `NaN` always return `false`.

Reply:
Agree that manual testing would be difficult. Is an automated test possible or necessary?

Reply:
I can probably rig something of an automated unit test in place to test the logic of its method.  It'd be harder to set up a proper integrated test for it.

My personal feeling is that it's _fiiine_.  Of course... that's often one of those "famous last words", so perhaps I should ignore that feeling here.

Reply:
I'll defer to your judgement on this as to whether it is worth the effort.